### PR TITLE
DataFrame.rolling documentation of method='table'

### LINKED ISSUE #40374: ENH: Rolling window apply to all columns and not just individual columns
#### Is your feature request related to a problem?

With a `group.apply` the applied function acts on all columns available to provide a result. With `rolling.apply` it applies the function to each column individually - it would be great if you could apply it to all the columns as well.

e.g. the following is possible with `groupby.apply`

```
df = pd.DataFrame({'A': range(10), 'B': range(10,0,-1)}).reset_index()
df['index'] = df['index'] % 3
df.groupby('index').apply(lambda df: df['A'] + df['B'])
```

However, you can't do the same equivalent with `rolling.apply`:

```
df = pd.DataFrame({'A': range(10), 'B': range(10,0,-1)}).reset_index()
df.rolling(3, on='index').apply(lambda df: (df['A'] * df['B']).sum())
```

Gives the error: `KeyError: 'A'`

This is because the `apply` function is applied to each column separately. The series is passed to the function not the dataframe. This feels inconsistent and annoying if you want to apply a function to the grouped rolling data. It would be great if there were a way to do it 😃

### PR DESCRIPTION
**Pull request by joshlk**
- [X] closes #40374 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -897,6 +897,30 @@ class Window(BaseWindow):
     2013-01-01 09:00:03  3.0
     2013-01-01 09:00:05  NaN
     2013-01-01 09:00:06  4.0
+
+    Rolling window apply function that uses mutliple columns as input using
+    method='table'. The apply function multiplies A with B and summs all rows
```
Comment (mzeitlin11) [inline pandas/core/window/rolling.py:902]:
```suggestion
    method='table'. The apply function multiplies A with B and sums all rows
```

File: pandas/core/window/rolling.py [hunk 2]
```diff
@@ -897,6 +897,30 @@ class Window(BaseWindow):
     2013-01-01 09:00:03  3.0
     2013-01-01 09:00:05  NaN
     2013-01-01 09:00:06  4.0
+
+    Rolling window apply function that uses mutliple columns as input using
```
Comment (mzeitlin11) [inline pandas/core/window/rolling.py:901]:
```suggestion
    Rolling window apply function that uses multiple columns as input using
```

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr, @joshlk! A couple quick suggestions

File: pandas/core/window/rolling.py [hunk 3]
```diff
@@ -897,6 +897,32 @@ class Window(BaseWindow):
...
+
+    >>> df
+       A  B
+    0  0  5
+    1  1  4
+    2  2  3
+    3  3  2
+    4  4  1
+
+    >>> apply_func = lambda M: (M[:,0] * M[:,1]).sum(axis=0)
```
Comment (mroeschke) [inline pandas/core/window/rolling.py:915]: Could you write this as a `def` instead?

File: pandas/core/window/rolling.py [hunk 4]
```diff
@@ -897,6 +897,32 @@ class Window(BaseWindow):
     2013-01-01 09:00:03  3.0
     2013-01-01 09:00:05  NaN
     2013-01-01 09:00:06  4.0
+
+    Rolling window apply function that uses multiple columns as input using
```
Comment (mroeschke) [inline pandas/core/window/rolling.py:901]: This might be better in the rolling apply docstring.

Comment (joshlk) [reply pandas/core/window/rolling.py:901]: The doc example is primarily about how to use the `method='table'` feature in the rolling function. Also the `apply` doc currently doesn't include any examples - all the examples seem to be in the rolling doc (e.g. `.sum(std=3)`)

Comment (mroeschke) [reply pandas/core/window/rolling.py:901]: Right but the particular usage of `method='table'` feature in the rolling function is only specific to `rolling(...).apply(...)`.

The `apply` doc could definitely use more examples!

Comment (joshlk) [reply pandas/core/window/rolling.py:901]: Would you like me to move this to the apply docstring then?

Comment (mroeschke) [reply pandas/core/window/rolling.py:901]: Yes, this example you made would go great in the rolling apply docstring

Comment (joshlk) [reply pandas/core/window/rolling.py]: @mroeschke Hi, I've tried adding the docs to the apply doc-string but I'm unsure how to do it - my latest commit isn't working. Sorry I am unfamiliar with how the doc-string is being generated.

Comment (mroeschke) [reply pandas/core/window/rolling.py]: You can see in this CI check what modifications need to be made to get this check to pass: https://github.com/pandas-dev/pandas/pull/40519/checks?check_run_id=2342425934

Additionally you can see here how to render the docs locally to see if there's inconsistent formatting somewhere: https://pandas.pydata.org/pandas-docs/dev/development/contributing_documentation.html#building-the-documentation

**Comment by simonjayhawkins** [comment]
@joshlk ci is failing can you fix-up

**Comment by joshlk** [comment]
> @joshlk ci is failing can you fix-up

Hi, I've tried adding the docs to the apply doc-string but I'm unsure how to do it - my latest commit isn't working. Sorry I am unfamiliar with how the doc-string is being generated. Could you please help

**Comment by datapythonista** [comment]
@joshlk can you have a look at the points of the previous comment, and try to leave the CI green please?

File: pandas/core/window/rolling.py [hunk 5]
```diff
@@ -1610,6 +1610,36 @@ def count(self):
...
         create_section_header("See Also"),
         template_see_also[:-1],
+        create_section_header("Examples"),
+        dedent(
+        """
+        Apply function that uses multiple columns as input using
+        method='table'. The apply function multiplies A with B and sums all rows
+        (dot product).
+        
+        >>> df = pd.DataFrame({{'A': range(5), 'B': range(5,0,-1)}})
```
Comment (Flix6x) [inline pandas/core/window/rolling.py:1620]: I get a `TypeError: unhashable type: 'dict'` when I run this line. When I replace `{{` with `{` and `}}` with `}` it works, though.

**Comment by mroeschke** [comment]
Thanks for the PR, but appears that progress has stalled. Closing, but let us know if you're still interested in working on this and we'd be happy to reopen.

**Pull request closed without merging**